components/proxy/src/main/java/com/hotels/styx/startup/BaseService.java

```diff
@@ -0,0 +1,85 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.startup;
+
+import com.google.common.util.concurrent.AbstractService;
+import org.slf4j.Logger;
+
+import java.util.concurrent.ExecutorService;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * A base to build services on. Adds concurrency, logging, error handling.
+ */
+public abstract class BaseService extends AbstractService {
```

Comment:
We already have a `StyxService` and `AbstractStyxService` abstraction. Please look into them. We should use the existing classes instead. Unless there are compelling reasons not to do otherwise?